cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -222,4 +227,51 @@ private void updateTetherStatus(HttpResponder responder, String peer, TetheringS
     }
     responder.sendStatus(HttpResponseStatus.OK);
   }
+
+  /**
+   * Processes the list of tethering program updates received from the client and updates the program statuses
+   */
+  private void processProgramUpdates(String content) {
+    if (content.isEmpty()) {
+      return;
+    }
+    TetheringProgramUpdate programUpdate = GSON.fromJson(content, TetheringProgramUpdate.class);
```

Comment:
Instead of doing this, it seems simpler to send over the list of Notifications without trying to interpret what they mean, and just write them as-is to the topic. This would preserve everything as-is and you wouldn't need to worry about what value to put for stopping, etc. Could we do that instead or is there a reason we need to go through the ProgramStateWriter?